src/Data/Aeson/TextMap.hs

```diff
@@ -0,0 +1,102 @@
+{-# LANGUAGE CPP #-}
+{-# LANGUAGE DeriveDataTypeable #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE TemplateHaskell #-}
+module Data.Aeson.TextMap (
+    TextMap,
+    difference,
+    empty,
+    foldrWithKey,
+    fromHashMap,
+    fromList,
+    fromListWith,
+    insert,
+    keys,
+    lookup,
+    mapKeyVal,
+    singleton,
+    size,
+    toList,
+    toAscList,
+    toHashMap,
+    traverseWithKey
+) where
+
+#if 1
+import Control.DeepSeq (NFData(..))
+import Data.Data (Data)
+import Data.Hashable (Hashable(..))
+import Data.HashMap.Strict (HashMap)
+import Data.List (sortBy)
+import Data.Ord (comparing)
+import Data.Text (Text, unpack)
+import Data.Typeable (Typeable)
+import Prelude hiding (lookup)
+import Data.Bifunctor (first)
+
+
+import qualified Data.HashMap.Strict as H
+import qualified Language.Haskell.TH.Syntax as TH
+
+newtype TextMap v = TextMap { unTextMap :: HashMap Text v }
+  deriving (Eq, Ord, Typeable, Data)
+  deriving newtype (Read, Show, Functor, Hashable, NFData)
+
+instance TH.Lift v => TH.Lift (TextMap v) where
+  lift (TextMap m) = [| TextMap (H.fromList . map (first pack) $ m') |]
+      where m' = map (first unpack) . H.toList $ m
```

Comment:
add `liftTyped = error "TODO"`. I'll fix it afterwards.

Let's make GHC-9.0 & 8.10 jobs happy, I'll take care of the rest + benchmark jobs.

Reply:
... or you can look at `Lift Value`, and do the same: https://hackage.haskell.org/package/aeson-1.5.6.0/docs/src/Data.Aeson.Types.Internal.html#line-440